pkg/utils/helpers.go

```diff
@@ -350,22 +350,25 @@ func IsOwnedByCMA(addon *addonapiv1alpha1.ManagedClusterAddOn) bool {
 	return false
 }
 
-// GetSpecHash returns the sha256 hash of the spec field of the given object
+// GetSpecHash returns the sha256 hash of the spec field or other config fields of the given object
 func GetSpecHash(obj *unstructured.Unstructured) (string, error) {
 	if obj == nil {
 		return "", fmt.Errorf("object is nil")
 	}
-	spec, ok := obj.Object["spec"]
-	if !ok {
-		return "", fmt.Errorf("object has no spec field")
-	}
 
-	specBytes, err := json.Marshal(spec)
+	// remove non config related fields, the remaining filed should be spec, data, other use defined fields or even empty
+	delete(obj.Object, "apiVersion")
```

Comment:
This will modify the input object. Is that going to cause any unintended issue? May be a better choice is to copy into a new map and use that calculate hash?

```
cp := map[string]any{}
for k, v := range obj.Object {
  switch k: {
    case "apiVersion", "kind", "metadata", "status":
    default:
       cp[k] = v
   }
} 
```

Reply:
Thanks for raising this, have modified the code.